### PR TITLE
"End Session/Quiz" Dialog Confirmation Appearing on Summary Pages

### PR DESCRIPTION
**Pull request by EricosEagle**
I noticed that I started to get a popup that blocks me from moving from the summary to the homepage, so I exempted it from the blocking function. I also replaced the regex with  `startsWith` / `endsWith` to make the code a bit clearer and faster.
Finally, I reduced the animation time of the wrapping up popup as I felt that it was a bit too long.

### CODE REVIEW

**Comment by salemlf** [comment]
I've been noticing this issue with the popup too! Not sure exactly how it was introduced. Navigation blocking is a pain, so the current solution is somewhat hacky. I'm in the process of migrating to a different router to solve this issue (and other problems), but this is great to fix in the meantime! I'll take a look over the changes you made 😊

File: src/pages/ReviewSession.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ const WrappingUpTxt = styled.h4`
   color: white;
 `;
 
-const animationDuration = 1.5;
+const animationDuration = 1.0;
```
Comment (salemlf) [inline src/pages/ReviewSession.tsx:72]: Would you mind creating a separate PR for this "Wrapping Up" animation duration? I know it's only a one-line change, but I generally try (keyword, "try" lol) to keep unrelated changes in separate PRs as it keeps things cleaner and is easier to revert if some changes end up causing a bug

File: src/components/LeaveSessionPrompt/LeaveSessionPrompt.tsx [hunk 2]
```diff
@@ -13,13 +13,13 @@ const blockUserLeavingPage = ({
...
-    subjDetailsRegex.test(nextLocation.pathname) ||
-    nextLocation.pathname === "/reviews/summary" ||
-    nextLocation.pathname === "/lessons/quiz" ||
-    nextLocation.pathname === "/lessons/summary"
+    // allowing user to view subjects pages during reviews and to review summary page
+    nextLocation.pathname.startsWith("/subjects") ||
+    // Don't block redirection to / from summaries
+    nextLocation.pathname.endsWith("/summary") ||
+    currentLocation.pathname.endsWith("/summary") ||
+    nextLocation.pathname.endsWith("/quiz")
```
Comment (salemlf) [inline src/components/LeaveSessionPrompt/LeaveSessionPrompt.tsx:22]: It looks like this has an issue where it doesn't account for navigating back from subject details page to a review or lesson session, see video:

https://github.com/salemlf/hakubun/assets/25971591/9e7c7cac-8c47-48ca-955c-da0322a075a3

However, seems like an easy fix! If you add an additional condition to check for navigating to a session: `|| nextLocation.pathname.endsWith("/session")`, this fixes the bug. After adding that condition: https://github.com/salemlf/hakubun/assets/25971591/6df37971-e178-44c0-8026-ea1b19737a6f

Comment (EricosEagle) [reply src/components/LeaveSessionPrompt/LeaveSessionPrompt.tsx:22]: Nice catch! I didn't run into that when I checked it but I'll add the fix now!

File: src/pages/ReviewSession.tsx (hunk 1 above)
Comment (EricosEagle) [reply src/pages/ReviewSession.tsx:72]: Sure thing!

**Comment by EricosEagle** [comment]
Hey, is there anything else I should change?
If not, I think this should be merged ASAP as the bug is very noticeable and disruptive.

**Comment by EricosEagle** [comment]
> Hey, is there anything else I should change?
> If not, I think this should be merged ASAP as the bug is very noticeable and disruptive.

An idea that came to mind is that rather than saying which paths we don't want to block, we can explicitly state which ones we do (from what I can tell, we only want to block moving back to the homepage from a lesson quiz or review session)

**Comment by EricosEagle** [comment]
> > Hey, is there anything else I should change?
> > If not, I think this should be merged ASAP as the bug is very noticeable and disruptive.
> 
> An idea that came to mind is that rather than saying which paths we don't want to block, we can explicitly state which ones we do (from what I can tell, we only want to block moving back to the homepage from a lesson quiz or review session), in order to increase performance. But as you are planning to migrate to a new router soon I'm not sure how much time I should invest.

Added this as a new commit, if you prefer this method I will remove the first commit, otherwise I will restore the code to what it was before.

**Comment by salemlf** [comment]
> > Hey, is there anything else I should change?
> > If not, I think this should be merged ASAP as the bug is very noticeable and disruptive.
> 
> An idea that came to mind is that rather than saying which paths we don't want to block, we can explicitly state which ones we do (from what I can tell, we only want to block moving back to the homepage from a lesson quiz or review session), in order to increase performance. But as you are planning to migrate to a new router soon I'm not sure how much time I should invest.

That's a good idea, only allowing specific paths totally makes sense! This is basically how I have it implemented in the version with the new router. I'm just finishing up migrating all tests to use the new router setup, so should have that all done within the week. I'll take a look at this PR after work, would be good to have a fix for this issue in the meantime 😊

**Review by salemlf — APPROVED** [review]

**Pull request merged**